File: src/main/java/io/renren/modules/sys/entity/SysRoleEntity.java
package io.renren.modules.sys.entity;


import lombok.Data;
import org.hibernate.validator.constraints.NotBlank;
import org.springframework.format.annotation.DateTimeFormat;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 角色
 *
 * @author ChunLiang Hu
 * @email davichi2009@gmail.com
 * @date 2016年9月18日 上午9:27:38
 */
@Data
public class SysRoleEntity implements Serializable {
    private static final long serialVersionUID = 1L;

    /**
     * 角色ID
     */
    private Long roleId;

    /**
     * 角色名称
     */
    @NotBlank(message = "角色名称不能为空")
    private String roleName;

    /**
     * 备注
     */
    private String remark;

    /**
     * 创建者ID
     */
    private Long createUserId;

    private List<Long> menuIdList;

    /**
     * 创建时间
     */
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;


}

File: src/main/java/io/renren/modules/sys/entity/SysRoleEntity.kt
package io.renren.modules.sys.entity


import org.hibernate.validator.constraints.NotBlank
import org.springframework.format.annotation.DateTimeFormat
import java.io.Serializable
import java.time.LocalDateTime

/**
 * 角色
 *
 * @author ChunLiang Hu
 * @email davichi2009@gmail.com
 * @date 2016年9月18日 上午9:27:38
 */
data class SysRoleEntity(

        /**
         * 角色ID
         */
        var roleId: Long? = null,

        /**
         * 角色名称
         */
        @NotBlank(message = "角色名称不能为空")
        var roleName: String? = null,

        /**
         * 备注
         */
        var remark: String? = null,

        /**
         * 创建者ID
         */
        var createUserId: Long? = null,

        var menuIdList: List<Long>? = null,

        /**
         * 创建时间
         */
        @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
        var createTime: LocalDateTime? = null,

        private val serialVersionUID: Long = 1L

) : Serializable
